model/map/src/main/java/org/keycloak/models/map/clientscope/MapClientScopeAdapter.java

```diff
@@ -72,8 +79,20 @@ public void setProtocol(String protocol) {
         entity.setProtocol(protocol);
     }
 
+    @Override
+    public RealmModel getRealm() {
+        return realm;
+    }
+
     @Override
     public void setAttribute(String name, String value) {
+        boolean valueUndefined = value == null || "".equals(value.trim());
+
+        if (valueUndefined) {
+            removeAttribute(name);
+            return;
+        }
```

Comment:
@hmlnarik  This change was done to make it consistent with `MapClientAdapter`. Originally it comes from https://issues.redhat.com/browse/KEYCLOAK-16676 for clients.  

I wonder if ti makes sense to have it here, if so there are some tests expecting empty value attributes to be returned and also `ClientScopeAdapter` would need to be updated. 

I'd like to know your opinion before I go ahead with this one, thank you.

Reply:
The broken test was brought in with #9327.
@dgozalo Any reason (apart from it being the current behaviour) for checking that [the `ClientScopeModel.DYNAMIC_SCOPE_REGEXP` attribute is empty](https://github.com/keycloak/keycloak/blob/e751626ac83515ef363ffde19d803185ab4d95d0/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/client/ClientScopeTest.java#L700)? It seems to me that it should have been removed.

Reply:
@vramik I believe making this behaviour consistent should be done in a separate issue across all the adapters. Could you please introduce it?

Reply:
https://github.com/keycloak/keycloak/issues/9741